modules/python/python-logmsg.c

```diff
@@ -276,6 +276,45 @@ py_log_message_get(PyLogMessage *self, PyObject *args, PyObject *kwrds)
   return default_value;
 }
 
+static PyObject *
+py_log_message_get_as_str(PyLogMessage *self, PyObject *args, PyObject *kwrds)
+{
+  const gchar *key = NULL;
+  Py_ssize_t key_len = 0;
+  PyObject *default_value = NULL;
+  const gchar *encoding = "utf-8";
+  const gchar *errors = "strict";
+
+  static const gchar *kwlist[] = {"key", "default", "encoding", "errors", NULL};
+  if (!PyArg_ParseTupleAndKeywords(args, kwrds, "z#|Oss", (gchar **) kwlist, &key, &key_len, &default_value,
+                                   &encoding, &errors))
+    return NULL;
+
+
+  NVHandle handle = log_msg_get_value_handle(key);
+  gssize value_len = 0;
+  LogMessageValueType type;
+  const gchar *value = log_msg_get_value_with_type(self->msg, handle, &value_len, &type);
+
+  if (!value || type == LM_VT_NULL)
+    {
+      if (!default_value)
+        Py_RETURN_NONE;
+
+      if (!PyUnicode_Check(default_value) && default_value != Py_None)
+        {
+          PyErr_Format(PyExc_TypeError, "default is not a string object");
+          return NULL;
+        }
+
+      Py_XINCREF(default_value);
+      return default_value;
+    }
+
+  APPEND_ZERO(value, value, value_len);
+  return PyUnicode_Decode(value, value_len, encoding, errors);
```

Comment:
If I understand correctly this function was originally imagined to solve the problem, that `str(b'some-bytes-data')` does not produce such elegant outputs as any other `str()` call for any other types.

I think that `PyUnicode_Decode()` should only be called when the type is `LM_VT_STRING`, for every other type we should return the same output as `str()` would, e.g.:
```python
>>> str(True)
'True'
>>> str(None)
'None'
>>> str(datetime.datetime(2022, 10, 4, 14, 48, 12, 123000))
'2022-10-04 14:48:12.123000'
```

Maybe the list could be a difference also, because we want to get rid of bytes there, too:
```python
>>> str([b'a', b'b', b'c'])
"[b'a', b'b', b'c']"
```
(`"['a', 'b', 'c']"` might look better)

I think the user is not interested in our internal representation of the values, like how we call true and false in our NV pairs, or how we store syslog-ng-like lists, but they would want to see an `str()` output for the non-bytes fields, and a unicode representation of bytes fields.

The `str()` call can be achieved easily with: https://docs.python.org/3/c-api/object.html#c.PyObject_Str

Reply:
So to clarify, this function does not convert a LogMessage value to a Python type and then try to decode it as a string. That would be unnecessarily expensive.

I get the LogMessage value, which is a `gchar*` (note that is can not be assumed that it is always UTF-8) and convert it to a Python string with PyUnicode_Decode() (the user must know the encoding).

I think the current behavior is correct.

Reply:
We've talked about this IRL and agreed to add a `repr='internal'` parameter to allow future implementation for a more Pythonic str method.